### PR TITLE
Appc 1275 add events purchase intent to sdk

### PR DESCRIPTION
**Pull request by nunomonteiro**
**What does this PR do?**

Adds a purchase_intent event to the sdk that gets fired when user launches billing flow
Changes the package name
from: com.appcoins.sdk.android.billing
to: com.appcoins.billing.sdk

**Database changed?**

No

**Where should the reviewer start?**

- [ ] EventLogger.java

**How should this be manually tested?**

 Open bds-sdk proj and run toolbox. Press launch billing flow button.
Take note of the time you did this. 10 min later you can ask @jose.pereira to check if your event showed up. Pro tip: you can force the sku value to a very specific value to ease the verification process

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1275](https://aptoide.atlassian.net/browse/APPC-1275)

**Questions:**
None

**Code Review Checklist**
- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            byte[] postData = jsonString.getBytes();
+            os.write(postData);
+            os.close();
+
+            connection.connect();
+
+            int code = connection.getResponseCode();
+            System.out.println(code);
+
+            try(BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream(), "utf-8"))){
```
Comment (brunovarelaAptoide) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java]: Code Style

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        jsonObj.put("data", dataObj);
+
+        LogEvent(eventName, jsonObj);
+    }
+
+    private void PostDataToURL(String urlStr, JSONObject jsonObj) {
+        URL url = null;
+        String responseStr = "";
+
+        try {
```
Comment (brunovarelaAptoide) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java]: Nested try catch

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java (hunk 1 above)
Comment (brunovarelaAptoide) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java]: Simplify the try. We have an IOException from the first try.

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import java.net.ProtocolException;
+import java.net.URL;
+
+import javax.net.ssl.HttpsURLConnection;
+
+public class EventLogger {
+
+    public static String ENTITY_NAME = "SDK";
+
+    private final String BASE_URL = "https://ws75.aptoide.com/api/7/";
```
Comment (brunovarelaAptoide) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java]: Pass this URL to build config

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java [hunk 4]
```diff
@@ -96,12 +101,18 @@ public LaunchBillingFlowResult launchBillingFlow(String skuType, String sku, Str
     }
     try {
 
+      String valueStr = ""; //TODO how to get the purchase value, why do we need it?
```
Comment (brunovarelaAptoide) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java]: Is this still in TODO?

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java [hunk 5]
```diff
@@ -12,17 +12,22 @@
...
   private final int apiVersion;
   private final String packageName;
   private WalletBillingService service;
   private boolean isServiceReady;
+  private EventLogger eventLogger;
 
   public AppCoinsAndroidBillingRepository(int apiVersion, String packageName) {
     this.apiVersion = apiVersion;
     this.packageName = packageName;
+    this.eventLogger = new EventLogger();
```
Comment (brunovarelaAptoide) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java]: Maybe create the EventLogger in CatapultBillingAppCoinsFactory and pass by parameter. We create all the Billing dependencies in CatapultBillingAppCoinsFactory.

**Review by brunovarelaAptoide — CHANGES_REQUESTED** [review]

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java (hunk 3 above)
Comment (nunomonteiro) [reply android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java]: I like the way of thinking of this suggestion but I think that this var is only relevant to eventLogger

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java (hunk 5 above)
Comment (nunomonteiro) [reply android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java]: Awesome point!

**Pull request merged**